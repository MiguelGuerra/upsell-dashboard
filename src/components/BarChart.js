import React from 'react'
import { Bar } from 'react-chartjs-2'

function BarChart({
    labelsArray,
    caption,
    dataArray,
    lineColor }) {
    const data = {
        labels: labelsArray,
        datasets: [
            {
                label: caption,
                data: dataArray,
                borderColor: lineColor,
                backgroundColor: lineColor,
                pointBackgroundColor: lineColor,
                pointBorderColor: lineColor
            }
        ]
    }
    return (
        <div className="bar-chart">
            <Bar
                data={data}
                options={{ maintainAspectRatio: false }} />
        </div>
    )
}

export default BarChart
