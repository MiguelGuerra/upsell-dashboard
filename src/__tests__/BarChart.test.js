import React from 'react'
import renderer from 'react-test-renderer'
import BarChart from '../components/BarChart'

it("BarChart renders when is receives props", () => {
    const tree = renderer.create(
        <BarChart
            labelsArray={['test 4', 'test 2']}
            caption="Test Caption"
            dataArray={['45', '50']}
            lineColor="#41a6c3"
        />).toJSON();
    expect(tree).toMatchSnapshot();
})