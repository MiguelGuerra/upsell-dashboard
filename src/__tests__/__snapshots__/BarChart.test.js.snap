// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart renders when is receives props 1`] = `
<div
  className="bar-chart"
  data-testid="bar-chart"
>
  <canvas
    data-testid="canvas"
    height={150}
    onClick={[Function]}
    role="img"
    width={300}
  />
</div>
`;
